handler/src/main/java/io/netty/handler/ssl/EnhancingX509ExtendedTrustManager.java

```diff
@@ -93,28 +103,65 @@ private static void throwEnhancedCertificateException(X509Certificate[] chain, C
             throws CertificateException {
         // Matching the message is the best we can do sadly.
         String message = e.getMessage();
-        if (message != null && e.getMessage().startsWith("No subject alternative DNS name matching")) {
-            StringBuilder names = new StringBuilder(64);
+        if (message != null &&
+                (message.startsWith("No subject alternative") || message.startsWith("No name matching"))) {
+            StringBuilder sb = new StringBuilder(128);
+            sb.append(message);
+            // Some exception messages from sun.security.util.HostnameChecker do not end with a dot.
+            if (message.charAt(message.length() - 1) != '.') {
+                sb.append('.');
+            }
+            sb.append(" Chain of ").append(chain.length).append(" certificate(s):");
             for (int i = 0; i < chain.length; i++) {
                 X509Certificate cert = chain[i];
                 Collection<List<?>> collection = cert.getSubjectAlternativeNames();
+                sb.append(' ').append(i + 1).append(". subjectAlternativeNames=[");
                 if (collection != null) {
+                    boolean hasNames = false;
                     for (List<?> altNames : collection) {
-                        // 2 is dNSName. See X509Certificate javadocs.
-                        if (altNames.size() >= 2 && ((Integer) altNames.get(0)).intValue() == 2) {
-                            names.append((String) altNames.get(1)).append(",");
+                        final int nameType = ((Integer) altNames.get(0)).intValue();
```

Comment:
is it guaranteed that `altNames` is not empty ? Maybe consider adding a check just to be safe... 


```suggestion
                        if (altNames.isEmpty()) {
                            continue;
                        }
                        final int nameType = ((Integer) altNames.get(0)).intValue();
```

Reply:
Good point, added a check to ensure it was at least 2 elements in the list